addons/website/static/tests/tours/snippet_table_of_content.js

```diff
@@ -38,4 +38,46 @@ wTourUtils.registerWebsitePreviewTour('snippet_table_of_content', {
     checkTOCNavBar(1, 0),
     scrollToHeading(3),
     checkTOCNavBar(1, 1),
+    wTourUtils.clickOnEdit(),
+    {
+        content: "Click on the first TOC's title",
+        trigger: 'iframe .s_table_of_content:eq(0) h1',
+    },
+    {
+        content: "Hide the first TOC on mobile",
+        trigger: '[data-toggle-device-visibility="no_mobile"]',
+    },
+    {
+        content: "Click on the second TOC's title",
+        trigger: 'iframe .s_table_of_content:eq(1) h1',
+    },
+    {
+        content: "Hide the second TOC on desktop",
+        trigger: '[data-toggle-device-visibility="no_desktop"]',
+    },
+    ...wTourUtils.clickOnSave(),
+    {
+        content: "Check that we have the good TOC on desktop",
+        trigger: 'iframe .s_table_of_content.o_snippet_mobile_invisible',
+        run: () => {}, // This is a check.
+    },
+    {
+        content: "Check that the mobile TOC is not visible on desktop",
+        trigger: 'iframe .s_table_of_content:not(.o_snippet_desktop_invisible)',
+        run: () => {}, // This is a check.
+    },
+    {
+        content: "Toggle de mobile view",
+        trigger: '.o_mobile_preview',
+    },
+    {
+        content: "Check that we have the good TOC on mobile",
+        trigger: 'iframe .s_table_of_content.o_snippet_desktop_invisible',
+        run: () => {}, // This is a check.
+    },
+    {
+        content: "Check that the desktop TOC is not visible on mobile",
+        trigger: 'iframe .s_table_of_content:not(.o_snippet_mobile_invisible)',
+        run: () => {}, // This is a check.
+    },
```

Comment:
I don't understand those steps. The o_snippet_XXX_invisible classes are not the ones who determine if one's is visible or not, but ok to check them in tests. However, what you do here does not seem ok for multiple reasons:
```javascript
    {
        content: "Check that the mobile TOC is not visible on desktop",
        trigger: 'iframe .s_table_of_content:not(.o_snippet_desktop_invisible)',
        run: () => {}, // This is a check.
    },
```
That trigger is always true no matter what. Of course there is a table_of_content that does not have this class if you do not target a specific table of content :thinking:
Correct me if I am wrong but those classes do not change whether or not we are in mobile mode/preview or not... so you basically are checking the same thing outside and in preview mode :thinking: In the end, either you toggle the mobile preview and you check the real visibility of the items on and off mobile view, or you solely rely on the o_snippet_XXX_invisible classes... but mixing both actually does not test this properly.